packages/api/src/Friend/Friend.entity.ts

```diff
@@ -0,0 +1,114 @@
+import { Field, InputType, ObjectType, registerEnumType } from '@nestjs/graphql';
+import { Column, Entity, PrimaryGeneratedColumn, Unique, CreateDateColumn, ManyToOne } from 'typeorm';
+
+import { CMBaseEntity } from '../lib/Base.entity';
+import { UserWithPassword, User } from '../User/User.entity';
+
+export enum FriendStatus {
+  pending = 'pending',
+  accepted = 'accepted',
+  rejected = 'rejected',
+}
+
+registerEnumType(FriendStatus, {
+  name: 'FriendStatus'
+});
+
+@ObjectType()
+@Entity('friend')
+@Unique('friend_rel', ['user1Id', 'user2Id'])
+export class Friend extends CMBaseEntity {
+  @PrimaryGeneratedColumn('uuid')
+  @Field()
+  id: string;
+
+  @Column()
+  @Field()
+  user1Id: string;
+
+  @Column()
+  @Field()
+  user2Id: string;
+
+  @CreateDateColumn()
+  @Field()
+  requested: Date;
+
+  @Column({ type: 'simple-enum', enum: FriendStatus, default: FriendStatus.pending })
+  @Field()
+  status: FriendStatus;
+
+  @Column()
+  @Field()
+  initiator: string;
+
+  @Column({ nullable: true })
+  @Field({ nullable: true })
+  since: Date;
+
+  @ManyToOne(() => UserWithPassword)
+  @Field(() => User)
+  user1: UserWithPassword;
+
+  @ManyToOne(() => UserWithPassword)
+  @Field(() => User)
+  user2: UserWithPassword;
+}
+
+@InputType()
+export class CreateFriendInput {
+  @Field()
+  fromId: string;
+
+  @Field()
+  toId: string;
+}
+
+@InputType()
+export class UpdateFriendInput {
+
+  @Field()
+  id: string;
+
+  @Field({ nullable: true })
```

Comment:
Is this really nullable?

Reply:
If we want to change a friendship from the admin, wouldn't be enough with the friendship id? Maybe I just want to change the status...

Reply:
change `respondToFriendRequest` input as user1Id, user2Id, response.